src/test/kotlin/com/terraformation/backend/species/db/SpeciesImporterTest.kt

```diff
@@ -659,29 +650,38 @@ internal class SpeciesImporterTest : DatabaseTest(), RunsAsUser {
                 modifiedTime = now),
         )
 
-    val actual = speciesDao.findAll()
-    assertEquals(expected.toSet(), actual.toSet())
+    assertEquals(expected.toSet(), speciesDao.findAll().map { it.copy(id = null) }.toSet())
     assertStatus(UploadStatus.Completed)
   }
 
   @Test
-  fun `importCsv rolls back changes and sets upload to failed if an error occurs`() {
+  fun `importCsv sets upload to failed if an error occurs`() {
+    val failingSpeciesStore = mockk<SpeciesStore>()
+    val failingImporter =
+        SpeciesImporter(
+            dslContext,
+            fileStore,
+            messages,
+            scheduler,
+            speciesChecker,
+            failingSpeciesStore,
+            uploadProblemsDao,
+            uploadsDao,
+            uploadService,
+            uploadStore,
+            userStore)
+    every { failingSpeciesStore.importSpecies(any(), any()) } throws RuntimeException("Failed!")
     every { fileStore.read(storageUrl) } returns
-        sizedInputStream("$header\nNew name,Common,Family,CR,false,Shrub,Recalcitrant,")
+        sizedInputStream("$header\nNew name,Common,Family,CR,false,Shrub,Recalcitrant,Tundra")
```

Comment:
This was just to make it closer to our real data, right? Or was it failing to fail without that?

Reply:
Yeah, just to make it closer to real.